docs/build/wasm/transaction-fees.md

```diff
@@ -0,0 +1,147 @@
+---
+sidebar_position: 2
+---
+
+# Transaction Fees
+
+## Weight
+
+As is also the case with Substrate, `pallet-contracts` uses [weightV2][weight] to charge execution fees. It is composed of `refTime` and `proofSize` :
+- refTime: The amount of computational time that can be used for execution, in picoseconds.
+- proofSize: The amount of storage space that can be used (also called storage bandwidth), in bytes. So access storage assume that it will grow the gas fees.
+
+:::info
+Gas = Weight = (refTime, proofSize)
+:::
+
+[Transaction Weight in Substrate Documentation][weight]
+
+## Storage Rent
+
+Storage rent, also called as *Automatic Deposit Collection* is a **mechanism** that ensure security of the chain by preventing on-chain storage spamming.
+It prevents malicious actors from spamming the network with low-value transactions and to ensure that callers have a financial stake when storing data on-chain.
+
+Users will be charged for every byte stored on-chain and the call will transfer this fee from the free balance of the user to the reserved balance of the contract. Note that the contract itself is unable to spend this reserved balance (but it can expose a function that remove on-chain storage and the caller will get the funds) .
+It also incentives users to remove unused data from the chain by getting rent fees back. Any user can get back the rent fees if they remove on-chain data (not specifically the user that was first charged for).
+
+### Storage Rent Calculation
+
+This fee is calculated with the price set for each storage item `DepositPerItem`, and for each byte of storage `DepositPerByte`. In Astar, the deposit fee are defined as follows (more detail is this [Astar forum post](https://forum.astar.network/t/revising-rent-fee-on-astar-shiden/4309/1)):
+
+| Deposit Type | Shiden               | Astar              |
+|--------------|----------------------|--------------------|
+| Per Storage Byte | 0.00002 SDN | 0.002 ASTR |
+| Per Storage Item |  	0.004 SDN   | 0.4 ASTR |
+
+##### Calculation
+
+When a user stores a new key/value in a `Mapping` field, one `DepositPerItem` is charged. The length in bytes of the value is added also added to the fee (so bytes length x `DepositPerByte`).     
+For example, if a user store a new entry in a `Mapping<u32, AccountId>` (`AccountId` is 32 bytes) it will be charged `DepositPerItem` + 32 x `DepositPerByte`. 
+
+### What does it mean ?
+
+#### For users
+
+The first call to a dApp (one or several's smart contracts) will usually be more expensive than the following ones.
+This is because the first call will create a lot of new entries for the `AccountId`. From the second call it should be way cheaper (or free) because it will just update those items.
```

Comment:
> This is because the first call will create a lot of new entries for the `AccountId`.

Doesn't have to be for the account calling the smart contract, it can essentially be any type of storage (e.g. a single value like `some_value = Option<HugeStruct>`

> From the second call it should be way cheaper (or free) because it will just update those items.

I'd suggest something along these lines:
"If consecutive calls only modify the existing database entry, caller is only charged for the extra bytes they add to the entry. In case they reduce the size of the DB entry, they will get refunded. What this means in practice is that user can increase their balance after interacting with a smart contract!"

Reply:
A bit modified, but added thanks